CHANGELOG.md

```diff
@@ -1,5 +1,90 @@
 # Changelog
 
+## [v0.4.0](https://github.com/delta-incubator/delta-kernel-rs/tree/v0.4.0/) (2024-10-22)
+
+[Full Changelog](https://github.com/delta-incubator/delta-kernel-rs/compare/v0.3.1...v0.4.0)
+
+**API Changes**
+
+*Breaking*
+
+1. `pub ScanResult.mask` field made private and only accessible as `ScanResult.raw_mask()` method [\#374]
+2. `ReaderFeatures` enum changes
+3. `WriterFeatures` enum changes
+4. `Error` enum changes
+5. Module moved: `mod delta_kernel::transaction` -> `mod delta_kernel::actions::set_transaction` [\#386]
+6. apply a schema to fix column names [\#331]
+7. change `default-feature` to be none [\#339]
+8. make `scan::execute` return a lazy iterator [\#340]
+9. filesystem client now always returns a sorted list [\#344]
+10. schema and expression FFI moved to their own `mod delta_kernel_ffi::schema` and `mod delta_kernel_ffi::expressions` [\#360]
+11. Parquet and JSON readers in `Engine` trait now take `Arc<Expression>` instead of `Expression` [\#364]
```

Comment:
```suggestion
11. Parquet and JSON readers in `Engine` trait now take `Arc<Expression>` (aliased to `ExpressionRef`) instead of `Expression` [\#364]
```